pages/api/auth/[...nextauth].ts

```diff
@@ -184,18 +184,26 @@ export const authOptions: NextAuthOptions = {
 
         if (token.sub) {
           const roles = await getTeamRoles(token.sub as string);
-          (session.user as any).roles = roles;
+          session.user.roles = roles;
         }
       }
 
       return session;
     },
+
+    async jwt({ token, user, trigger, session }) {
+      if (trigger === 'update') {
+        return { ...token, ...session.user };
```

Comment:
It looks like I can update any values in the session object by passing them from the client side. Since this object comes from the client, must be cautious while updating the session with those values. Make sure you only update the name in the session object. https://next-auth.js.org/getting-started/client#updating-the-session

Reply:
I intended for this callback to be able to manage any kind of session object update

Reply:
There is a problem allowing this way; the user can update any value in the session object. For example, if you have a role in the session object, the user can update the role to admin and gain access to admin pages. I think we should only allow updating the name in the session object. Let me know if you understand.